util/patch/patch.go

```diff
@@ -192,39 +205,73 @@ func (h *Helper) patchStatus(ctx context.Context, obj client.Object) error {
 //
 // Condition changes are then applied to the latest version of the object, and if there are
 // no unresolvable conflicts, the patch is sent again.
-func (h *Helper) patchStatusConditions(ctx context.Context, obj client.Object, forceOverwrite bool, ownedConditions []clusterv1.ConditionType) error {
-	// Nothing to do if the object isn't a condition patcher.
-	if !h.isConditionsSetter {
+func (h *Helper) patchStatusConditions(ctx context.Context, obj client.Object, forceOverwrite bool, ownedConditions []clusterv1.ConditionType, ownedV1beta2Conditions []string) error {
+	// Nothing to do if the object doesn't have conditions.
+	if len(h.clusterv1ConditionsFields) == 0 && len(h.metav1ConditionsFields) == 0 {
 		return nil
 	}
 
-	// Make sure our before/after objects satisfy the proper interface before continuing.
-	//
-	// NOTE: The checks and error below are done so that we don't panic if any of the objects don't satisfy the
-	// interface any longer, although this shouldn't happen because we already check when creating the patcher.
-	before, ok := h.beforeObject.(conditions.Getter)
-	if !ok {
-		return errors.Errorf("object %s doesn't satisfy conditions.Getter, cannot patch", before.GetObjectKind())
-	}
-	after, ok := obj.(conditions.Getter)
-	if !ok {
-		return errors.Errorf("object %s doesn't satisfy conditions.Getter, cannot patch", after.GetObjectKind())
-	}
-
-	// Store the diff from the before/after object, and return early if there are no changes.
-	diff, err := conditions.NewPatch(
-		before,
-		after,
-	)
-	if err != nil {
-		return errors.Wrapf(err, "object can not be patched")
+	// If the object has clusterv1 conditions, create a function applying corresponding changes if any.
+	var clusterv1ApplyPatch func(client.Object) error
+	if len(h.clusterv1ConditionsFields) > 0 {
+		// Make sure our before/after objects satisfy the proper interface before continuing.
+		//
+		// NOTE: The checks and error below are done so that we don't panic if any of the objects don't satisfy the
+		// interface any longer, although this shouldn't happen because we already check when creating the patcher.
+		before, ok := h.beforeObject.(conditions.Getter)
+		if !ok {
+			return errors.Errorf("object %s doesn't satisfy conditions.Getter, cannot patch", before.GetObjectKind())
```

Comment:
```suggestion
			return errors.Errorf("%s %s doesn't satisfy conditions.Getter, cannot patch", h.gvk.Kind, klog.KObj(before))
```
(also fine to do this in a follow-up, would be just nice to improve this here and in all similar cases below)

Reply:
done